### PR TITLE
initialize user service

### LINKED ISSUE #11: initialize login service with mvc pattern
We need a login service, initialize this service.

This service needs all the required endpoints for `access` and `refresh` tokens as well as the authentication middleware.

### PR DESCRIPTION
**Pull request by PaulKalho**
Closes #11

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,40 @@
...
+| spark-worker      |      |
+| spark-master      | 8080 |
+| airflow-webserver | 3333 |
+| airflow-cli       |      |
+| airflow-triggerer |      |
+| airflow-worker    |      |
+| airflow-scheduler |      |
+| postgres-airflow  |      |
+| redis-airflow     |      |
+
```
Comment (use-to) [inline README.md:36]: Weird doc

File: core/Dockerfile [hunk 2]
```diff
@@ -30,4 +30,4 @@ COPY . /app
 
 WORKDIR /app
 
-ENTRYPOINT ["uvicorn", "main:app", "--proxy-headers"]
+ENTRYPOINT ["uvicorn", "src.main:app", "--proxy-headers"]
```
Comment (use-to) [inline core/Dockerfile:33]: `src` wtf?

File: core/README.md [hunk 3]
```diff
@@ -1 +1,47 @@
...
+
+For local development create an `.env` file inside the `/src` directory.
+As an example see `src/.env.example`
+
+You can now start the uvicorn server locally with
+
+```sh
+uvicorn src.main:app --reload --port <port>
+```
+
```
Comment (use-to) [inline core/README.md:13]: look above

File: README.md [hunk 4]
```diff
@@ -1,6 +1,40 @@
...
+## Development
+
+You can find the development READMEs in the according directories
+
```
Comment (use-to) [inline README.md:40]: add env table

File: core/src/services/user_service/controllers/auth_controller.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                "iat": now,
+                "exp": now + timedelta(minutes=ENV.JWT_ACCESS_TOKEN_EXPIRE_MIN)
+            }) 
+       
+        refresh_token = create_token({
+                "iat": now,
+                "exp": now + timedelta(days=ENV.JWT_REFRESH_TOKEN_EXPIRE_DAYS)
+            })
+    else:
+        raise HTTPException(401, detail="wrong password")
```
Comment (use-to) [inline core/src/services/user_service/controllers/auth_controller.py:40]: fail early

File: core/src/services/user_service/controllers/auth_controller.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+    return (access_token, refresh_token)
```
Comment (use-to) [inline core/src/services/user_service/controllers/auth_controller.py:42]:
```suggestion
    return access_token, refresh_token
```

File: core/src/services/user_service/controllers/create_user.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
+from uuid import UUID, uuid4
+import bcrypt
+
+from src.utils.database.session_injector import get_database
+from src.services.user_service.models.user import User
+
+
+def createUser(email: str, password: str) -> UUID:
```
Comment (use-to) [inline core/src/services/user_service/controllers/create_user.py:8]: snake_case?

File: core/src/services/user_service/views/user.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
+from fastapi import APIRouter, HTTPException
+from src.utils.errors.error import handle_error
+
+from src.services.user_service.schemas.user import (CreateUserResponse, CreateUserRequest, LoginRequest, LoginResponse)
```
Comment (use-to) [inline core/src/services/user_service/views/user.py:4]: ?
```suggestion
from src.services.user_service.schemas.user import CreateUserResponse, CreateUserRequest, LoginRequest, LoginResponse
```

File: core/src/services/user_service/views/user.py [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+import src.services.user_service.controllers.create_user as create_user_controller 
+import src.services.user_service.controllers.auth_controller as auth_controller                                                
+
+router = APIRouter(prefix="/user", tags=["user"])
+
+@router.post("/create", response_model=CreateUserResponse)
+async def createUser(data: CreateUserRequest):
```
Comment (use-to) [inline core/src/services/user_service/views/user.py:12]: snake case

File: core/src/utils/config/environment.py [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+from typing import List
+
+class Settings(BaseSettings):
+    DATABASE_HOST: str = "core-postgres"
+    DATABASE_NAME: str = "core"
+    DATABASE_USER: str = "core"
+    DATABASE_PASSWORD: str = "core"
+    DATABASE_PORT: int = 5432
+    EMAIL_WHITELIST: List[str] = ["time.rwth-aachen.de"]
```
Comment (use-to) [inline core/src/utils/config/environment.py:11]: EMAIL_DOMAIN_WHITELIST

File: docker-compose.yml [hunk 11]
```diff
@@ -235,9 +235,13 @@ services:
   core:
     build: core
     restart: always
+    ports:
+      - "4000:4000"
   
-  core-postgres:
+  core-postgres:
```
Comment (use-to) [inline docker-compose.yml:241]:
```suggestion
  core-postgres:
```

**Review by use-to — CHANGES_REQUESTED** [review]

File: core/.dockerignore [hunk 12]
```diff
@@ -0,0 +1 @@
+.env
```
Comment (use-to) [inline core/.dockerignore:1]:
```suggestion
.env
.env*
```

File: core/Dockerfile [hunk 13]
```diff
@@ -30,4 +30,4 @@ COPY . /app
 
 WORKDIR /app
 
-ENTRYPOINT ["uvicorn", "main:app", "--proxy-headers"]
+CMD alembic upgrade head && uvicorn main:app --proxy-headers
```
Comment (use-to) [inline core/Dockerfile:33]: Weird why? ENTRYPOINT makes sense to me, but we can also do a CMD here. Why did you change that?

File: core/utils/database/import_models.py [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+import os
+
+
+def import_all_models(module_name: str) -> None:
+    package = importlib.import_module(module_name)
+    package_dir = os.path.dirname(package.__file__)
+
+    for _, name, is_pkg in pkgutil.iter_modules([package_dir]):
+        if not is_pkg:
+            importlib.import_module(f"{module_name}.{name}")
```
Comment (use-to) [inline core/utils/database/import_models.py:12]: Lets do this manually.

**Review by use-to — CHANGES_REQUESTED** [review]

File: core/Dockerfile (hunk 13 above)
Comment (PaulKalho) [reply core/Dockerfile:33]: Alternative would be `ENTRYPOINT ["sh", "-c", "alembic upgrade head &&  uvicorn main:app --proxy-headers"]`, or am I missing something? Seemed more natural using CMD here.

**Review by use-to — COMMENTED** [review]
LGTM

**Review by use-to — APPROVED** [review]
